packages/dds/task-manager/src/taskManager.ts

```diff
@@ -764,16 +770,15 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
      * Will replace all instances of the placeholderClientId with the current clientId. This should only be called when
      * transitioning from detached to attached and this.runtime.clientId is defined.
      */
-    private replacePlaceholderFromAllQueues() {
-        assert(this.clientId !== undefined, "this.runtime.clientId should be defined");
-        for (const taskId of this.taskQueues.keys()) {
-            const clientQueue = this.taskQueues.get(taskId);
+    private replacePlaceholderInAllQueues() {
+        assert(this.runtime.clientId !== undefined, "this.runtime.clientId should be defined");
+        for (const clientQueue of this.taskQueues.values()) {
             if (clientQueue === undefined) {
                 continue;
             }
             const clientIdIndex = clientQueue.indexOf(placeholderClientId);
             if (clientIdIndex !== -1) {
-                clientQueue[clientIdIndex] = this.clientId;
+                clientQueue.splice(clientIdIndex, 1, this.runtime.clientId);
```

Comment:
Nit: Does this need to use splice instead of just clientQueue[clientIdIndex] = this.runtime.clientId?

Reply:
Yes, I think I confused myself on this one 😅